internal/controller/integrationpipeline/integrationpipeline_adapter.go

```diff
@@ -99,15 +99,6 @@ func (a *Adapter) EnsureStatusReportedInSnapshot() (controller.OperationResult,
 		return controller.RequeueWithError(fmt.Errorf("failed to update test status in snapshot: %w", err))
 	}
 
-	// Remove the finalizer from Integration PLRs only if they are related to Snapshots created by Push event
-	// If they are related, then the statusreport controller removes the finalizers from these PLRs
-	if gitops.IsSnapshotCreatedByPACPushEvent(a.snapshot) && (h.HasPipelineRunFinished(a.pipelineRun) || pipelinerunStatus == intgteststat.IntegrationTestStatusDeleted) {
```

Comment:
I think here we still need some code to remove finalizer from integration test PLR for `override` snapshot because it has no chance to remove it in other place. Like `if gitops.IsOverrideSnapshot(a.snapshot) {`

Reply:
Others look good to me. Thanks!

Reply:
How's this?

`	if gitops.IsOverrideSnapshot(a.snapshot) && (h.HasPipelineRunFinished(a.pipelineRun) || pipelinerunStatus == intgteststat.IntegrationTestStatusDeleted) {
		err = h.RemoveFinalizerFromPipelineRun(a.context, a.client, a.logger, a.pipelineRun, h.IntegrationPipelineRunFinalizer)
		if err != nil {
			return controller.RequeueWithError(fmt.Errorf("failed to remove the finalizer: %w", err))
		}
	}`

Reply:
It look good to me.

Reply:
Done